### PR TITLE
Fix sorting of newly created case notes in records history

### PR DESCRIPTION
**Pull request by chubberlisk**
### *What is the problem we're trying to solve*

For newly created case notes, they appear at the bottom of the records history when they should appear at the top. This is because date of event was missing from the request which meant there was no date to sort by.

Also when testing this in staging, we also discovered a smol bug with the date of birth causing an issue when creating a case note with a null DOB.

### *What changes have we introduced*

This PR:

- fixes the null date of birth by doing a null check when doing `To.String()`
- makes sure we use date of event when creating a case note
- allows sorting for ISO format date of event

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

Check that these changes fix the issues in staging and add tests for `ToEntity` for `CreateCaseNoteRequest`.

## Link to JIRA ticket

https://hackney.atlassian.net/browse/SCS-592

### CODE REVIEW

File: SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs [hunk 1]
```diff
@@ -175,7 +175,8 @@ public static CaseNotesDocument ToEntity(this CreateCaseNoteRequest request)
             GenericCaseNote note = new GenericCaseNote()
             {
                 Timestamp = DateTime.Now.ToString("dd/MM/yyy hh:mm"),
-                DateOfBirth = request.DateOfBirth.ToString(),
+                DateOfBirth = request.DateOfBirth?.ToString(),
```
Comment (Renny4Real) [inline SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs:178]: Do we want to return the Datetime properties here as ISO format?

**Review by Renny4Real — APPROVED** [review]
👍🏾 Looks good

File: SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs [hunk 2]
```diff
@@ -240,19 +240,21 @@ public IOrderedEnumerable<CareCaseData> SortData(string sortBy, string orderBy,
...
+                    if (success) return timeStamp;
                 }
                 else
                 {
                     bool success = DateTime.TryParseExact(x.DateOfEvent, "dd/MM/yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateOfEvent);
-                    if (success) dt = dateOfEvent;
+                    if (success) return dateOfEvent;
+
+                    bool successForISOFormat = DateTime.TryParseExact(x.DateOfEvent, "O", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateOfEventISOFormat);
+                    if (success) return dateOfEventISOFormat;
```
Comment (JohnFarrellDev) [inline SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs:254]: Should this be `if (successForISOFormat)`

Comment (LBHTKarki) [reply SocialCareCaseViewerApi/V1/Gateways/ProcessDataGateway.cs]: well spotted, thank you! fixed

File: SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs (hunk 1 above)
Comment (LBHTKarki) [reply SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs]: We have aligned this with existing data i.e. used dd/mm/yyyy format

**Review by JohnFarrellDev — APPROVED** [review]
Looks good

**Review by Renny4Real — APPROVED** [review]

**Pull request merged**